lib/std/base64.zig

```diff
@@ -98,6 +99,31 @@ pub const Base64Encoder = struct {
         }
     }
 
+    // dest must be compatible with std.io.Writer's writeAll interface
+    pub fn encodeWriter(encoder: *const Base64Encoder, dest: anytype, source: []const u8) !void {
+        var temp = [_]u8{0} ** 5;
+        var chunker = window(u8, source, 3, 3);
+        while (chunker.next()) |chunk| {
+            const s = encoder.encode(&temp, chunk);
+            try dest.writeAll(s);
+        }
+    }
```

Comment:
Couple suggestions, limit variables to the smallest scope as possible, and make use of `undefined` when applicable, i.e.

```suggestion
    pub fn encodeWriter(encoder: *const Base64Encoder, dest: anytype, source: []const u8) !void {
        var chunker = window(u8, source, 3, 3);
        while (chunker.next()) |chunk| {
            var temp: [5]u8 = undefined;
            const s = encoder.encode(&temp, chunk);
            try dest.writeAll(s);
        }
    }
```

Reply:
Fair points, guess still have bad habit from C.